e2e_tests/tests/cluster/test_logging.py

```diff
@@ -105,13 +105,36 @@ def task_log_fields(follow: Optional[bool] = None) -> Iterable[LogFields]:
         return bindings.get_TaskLogsFields(session, taskId=task_id, follow=follow)
 
     try:
-        check_logs(
-            log_regex,
-            functools.partial(api.task_logs, session, task_id),
-            functools.partial(bindings.get_TaskLogsFields, session, taskId=task_id),
-        )
-    except socket.timeout:
-        raise TimeoutError(f"timed out waiting for {task_type} with id {task_id}")
+        result: Any = None
+
+        def do_check_logs() -> None:
+            nonlocal result
+            try:
+                check_logs(
+                    log_regex,
+                    functools.partial(api.task_logs, session, task_id),
+                    functools.partial(bindings.get_TaskLogsFields, session, taskId=task_id),
+                )
+                # Success!
+                result = True
+            except Exception as e:
+                result = e
+
+        thread = threading.Thread(target=do_check_logs, daemon=True)
+        thread.start()
+        thread.join(timeout=5 * 60)
```

Comment:
small nit that this stuffs configuration into the implementation.

Options I see, none of which I love (so indeed, maybe this is better):
* set a custom mark and in the test pull it out of `request.node.keywords`
* abuse `pytest.parameterize` to set a fixture

Reply:
honestly I'm inclined to leave it as-is.  `request.node.keywords` is something I've never heard of, which makes me think it would be a lot less readable.

A strategy with 'abuse' in the bullet point sounds undesirable.

My take is that putting config into the implementation is the least evil.